docs/user-guide/ingress.md

```diff
@@ -234,7 +265,7 @@ You can achieve the same by invoking `kubectl replace -f` on a modified Ingress
 
 ## Future Work
 
-* Various modes of HTTPS/TLS support (edge termination, sni etc)
+* Various modes of HTTPS/TLS support (eg: SNI, re-encryption)
```

Comment:
The first step is figuring out how we fold tcp into the Ingress which currently only supports L7 protocols, but there's been discussion around this if you're interested in participating (https://github.com/kubernetes/kubernetes/issues/19333).

Simply adding support to either the existing nginx controller (https://github.com/kubernetes/contrib/tree/master/ingress/controllers/nginx-third-party) or haproxy service loadbalancer (https://github.com/kubernetes/contrib/tree/master/service-loadbalancer) shouldn't be hard, because we can already mux on hostname as described in https://github.com/kubernetes/kubernetes/issues/19333.